src/test/scala/karazin/scala/users/group/week2/homework/HomeworkSpecification.scala

```diff
@@ -38,40 +39,40 @@ object HomeworkSpecification extends Properties("Homework"):
   }
 
   property("less or equal") = forAll { (left: Rational, right: Rational) =>
-    (left <= right) == ( left < right || left == right)
+    (left <= right) == (left < right || left == right)
   }
 
   property("greater") = forAll { (left: Rational, right: Rational) =>
     (left > right) == !(left <= right)
   }
 
   property("greater or equal") = forAll { (left: Rational, right: Rational) =>
-    (left >= right) == ( left > right || left == right)
+    (left >= right) == (left > right || left == right)
   }
 
   property("negation") = forAll { (rational: Rational) =>
-    ???
+    (-rational).toDecimalFormat().round == Rational(rational.numer * (-1), rational.denom).toDecimalFormat().round

```

Comment:
It's a dirty hack.
Please compare `num`s and `denom`s without casting to double.